vignettes/flare-metadata-example.Rmd

```diff
@@ -20,537 +23,313 @@ library(tidyr)
-time <- as.POSIXct(time, origin = '1970-01-01 00:00.00 UTC', tz = local_tzone)
-time <-  strftime(time , "%Y-%m-%dT%H:%M:%S%z")
-
-#States
-temperature <- ncvar_get(nc, "temp")
-oxygen <- ncvar_get(nc, "OXY_oxy")
-zone1temp <- ncvar_get(nc, "zone1temp")
-zone2temp <- ncvar_get(nc, "zone2temp")
-sw_factor <- ncvar_get(nc, "sw_factor")
-inflow_factor <- ncvar_get(nc, "inflow_factor")
-
-#Forecast timings
-forecasted <- ncvar_get(nc, "forecasted")
-data_assimilation <- if_else(forecasted == 0, 1, 0)
-#This doesn't work but we need to update our writing script to just have the date
-forecast_issue_time <- ncatt_get(nc, varid=0, attname = "history") 
-
-forecast_issue_time <- "2020-02-13T21:32:52-05:00"
-
-nc_close(nc)
+abstract_text <- "This is where a longer description of the forecast can be added"
+forecast_title <- "FLARE"
+intellectualRights <- "insert license"
+model_description <- list(
+  name = "General Lake Model",
+  type = "process-based",
+  repository = "https://github.com/AquaticEcoDynamics/GLM/releases/tag/v3.1.0"
+)
+me <- list(individualName = list(givenName = "Quinn", 
+                                 surName = "Thomas"),
+           electronicMailAddress = "rqthomas@vt.edu",
+           id = "https://orcid.org/0000-0003-1282-7825")
 ```
 
-## Saving to a standardized output format (Option 1)
+## Building the metadata
 
-Second, build a data frame for all variables with a depth dimension
+Open and extract information for building the metadata file
 
 ```{r}
-
-n_depths <- length(depth)
-state_names <- c("temperature", "oxygen")
-states <- list(temperature, oxygen)
-n_states <- length(state_names)
-df_combined <- list()
-
-for(k in 1:n_states){
-for(i in 1:n_depths){
-    df <- as_tibble(states[[k]][, ,i])
-    names(df) <- as.character(seq(1, ncol(states[[k]][, ,i])))
-    df <- cbind(time, df, data_assimilation)
-    df <- df %>% 
-      pivot_longer(cols = -c(time,data_assimilation), 
-                   names_to = "ensemble", 
-                   values_to = state_names[k]) %>% 
-      mutate(ensemble = as.integer(ensemble)) %>% 
-      mutate(depth = depth[i])
-    if(i == 1){
-    running_df <- df
-    }else{
-      running_df <- rbind(running_df, df)
-    }
+nc <- nc_open(file_name)
+t <- ncvar_get(nc,'time')  #Saved as seconds since 1970-01-01 00:00.00 UTC
+time <- as.POSIXct(t, 
+                   origin = '1970-01-01 00:00.00 UTC', 
+                   tz = "UTC")
+
+#data assimilation flag
+data_assimilation <- ncvar_get(nc, "data_assimilation")
+
+#These are in the netcdf global attributed
+forecast_issue_time <-ncatt_get(nc, varid = 0)$forecast_issue_time
+forecast_iteration_id <- ncatt_get(nc, varid = 0)$forecast_iteration_id
+forecast_project_id <- ncatt_get(nc, varid = 0)$forecast_project_id
+forecast_model_id <- ncatt_get(nc, varid = 0)$forecast_model_id
+if (is.null(forecast_model_id))
+  forecast_model_id <- "3663f9580cc7003542d538a2c96fdb4550a32bef" # github hash from https://github.com/CareyLabVT/FLARE
+
```

Comment:
This is the hacked model id

Reply:
perhaps comment shoul've linked to the commit id on gh: https://github.com/CareyLabVT/FLARE/commit/3663f9580cc7003542d538a2c96fdb4550a32bef